spinnaker/resource_application.go

```diff
@@ -51,16 +87,18 @@ type applicationRead struct {
 	} `json:"attributes"`
 }
 
-func resourceApplicationCreate(data *schema.ResourceData, meta interface{}) error {
+func resourceApplicationCreateOrUpdate(data *schema.ResourceData, meta interface{}) error {
 	clientConfig := meta.(gateConfig)
 	client := clientConfig.client
 	application := data.Get("application").(string)
 	email := data.Get("email").(string)
 	description := data.Get("description").(string)
 	platform_health_only := data.Get("platform_health_only").(bool)
 	platform_health_only_show_override := data.Get("platform_health_only_show_override").(bool)
+	cloud_providers := data.Get("cloud_providers").([]interface {})
+	permissions := data.Get("permissions").(interface {})
```

Comment:
You're asserting the type as interface{} here then *schema.Set on line 101 is there any reason you don't just do something like this here?:
```
if permissions, ok := data.Get("permissions").(*schema.Set); !ok {
  some error handling
}
```

Then the error check on line 101 is also only for the api.CreateOrUpdateApplication method not also the type assertion.

Reply:
Eliminated unneeded cast